flang/unittests/Optimizer/KindMappingTest.cpp

```diff
@@ -0,0 +1,144 @@
+//===- KindMappingTest.cpp ------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "flang/Optimizer/Support/KindMapping.h"
+#include "gtest/gtest.h"
+#include <string>
+
+using namespace fir;
+namespace llvm {
+struct fltSemantics;
+} // namespace llvm
+
+namespace mlir {
+class MLIRContext;
+} // namespace mlir
+
+using Bitsize = fir::KindMapping::Bitsize;
+using LLVMTypeID = fir::KindMapping::LLVMTypeID;
+
+struct DefaultStringTests : public testing::Test {
+public:
+  void SetUp() { defaultString = new KindMapping(context); }
+  void TearDown() { delete defaultString; }
+
+  KindMapping *defaultString{};
+  mlir::MLIRContext *context{};
+};
+
+struct commandLineStringTests : public testing::Test {
+public:
+  void SetUp() {
+    commandLineString = new KindMapping(
+        context, "i10:80,l3:24,a1:8,r54:Double,c20:X86_FP80,r11:PPC_FP128");
```

Comment:
Can you make the test use all of the floating-point semantic names for completeness?